siriuspy/siriuspy/diagsys/lidiag/pvs.py

```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
         return {'value': value}
 
 
+class LIPUStatusPV:
+    """LI PU Status PV."""
+
+    BIT_DISCONN = 0b000000000000000001
+    BIT_RUNSTOP = 0b000000000000000010
+    BIT_PREHEAT = 0b000000000000000100
+    BIT_CHRALLW = 0b000000000000001000
+    BIT_TRGALLW = 0b000000000000010000
+    BIT_EMRSTOP = 0b000000000000100000
+    BIT_CPS_ALL = 0b000000000001000000
+    BIT_THYHEAT = 0b000000000010000000
+    BIT_KLYHEAT = 0b000000000100000000
+    BIT_LVRDYOK = 0b000000001000000000
+    BIT_HVRDYOK = 0b000000010000000000
+    BIT_TRRDYOK = 0b000000100000000000
+    BIT_SELFFLT = 0b000001000000000000
+    BIT_SYS_RDY = 0b000010000000000000
+    BIT_TRGNORM = 0b000100000000000000
+    BIT_PLSCURR = 0b001000000000000000
+    BIT_VLTDIFF = 0b010000000000000000
+    BIT_CRRDIFF = 0b100000000000000000
+
+    PV_RUNSTOP = 0
+    PV_PREHEAT = 1
+    PV_CHRALLW = 2
+    PV_TRGALLW = 3
+    PV_EMRSTOP = 4
+    PV_CPS_ALL = 5
+    PV_THYHEAT = 6
+    PV_KLYHEAT = 7
+    PV_LVRDYOK = 8
+    PV_HVRDYOK = 9
+    PV_TRRDYOK = 10
+    PV_SELFFLT = 11
+    PV_SYS_RDY = 12
+    PV_TRGNORM = 13
+    PV_PLSCURR = 14
+    PV_VLTDIFF = 15
+    PV_CRRDIFF = 16
+
+    def compute_update(self, computed_pv, updated_pv_name, value):
+        """Compute Status PV."""
+        value = 0
+
+        # connected?
+        disconnected = \
+            not computed_pv.pvs[LIPUStatusPV.PV_RUNSTOP].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_PREHEAT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CHRALLW].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRGALLW].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_EMRSTOP].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CPS_ALL].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_THYHEAT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_KLYHEAT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_LVRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_HVRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SELFFLT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SYS_RDY].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRGNORM].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_PLSCURR].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_VLTDIFF].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CRRDIFF].connected
+        if disconnected:
```

Comment:
actually here you just neet to set all bits to 1, right? you could just do: `value = 2**(NBITS+1)-1`